hathor/transaction/transaction.py

```diff
@@ -260,6 +263,33 @@ def to_json(self, decode_script: bool = False, include_metadata: bool = False) -
         json['tokens'] = [h.hex() for h in self.tokens]
         return json
 
+    def verify_basic(self, skip_block_weight_verification: bool = False) -> None:
+        if self.is_genesis:
+            # TODO do genesis validation?
+            return
+        self.verify_without_storage()
+        self.verify_parents_basic()
+        self.verify_weight()
+
+    def verify_parents_basic(self) -> None:
```

Comment:
Shouldn't it be part of the `verify_without_storage()`?

Reply:
Maybe, these verifications already exist in other forms outside of the usual `verify` chain. They are integrated into the usual `verify` chain on the next branches, but I'm not sure if I can pull that integration yet (there are some restrictions on when some data is available, or on the order that I can verify transacions). I have to confirm if I can pull these changes.

Reply:
I checked it out, and it makes sense.

Reply:
I tried moving it and a bunch of tests broke. While I think the move makes sense and it might actually indicate something wrong on the tests, at this point I rather do this on a new PR.